### ISSUE TITLE
[1.12.0-rc2] `docker plugin install` fails if /var/lib/docker is a symlink

### ISSUE THREAD

**Issue by AkihiroSuda** [docker/docker#23753]
**Output of `docker version`:**

```
Client:
 Version:      1.12.0-rc2
 API version:  1.24
 Go version:   go1.6.2
 Git commit:   906eacd-unsupported
 Built:        Fri Jun 17 21:12:28 2016
 OS/Arch:      linux/amd64
 Experimental: true

Server:
 Version:      1.12.0-rc2
 API version:  1.24
 Go version:   go1.6.2
 Git commit:   906eacd-unsupported
 Built:        Fri Jun 17 21:12:28 2016
 OS/Arch:      linux/amd64
 Experimental: true
```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 1.12.0-rc2
Storage Driver: overlay2
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge null host overlay
Swarm: inactive
Runtimes: default
Default Runtime: default
Security Options: apparmor seccomp
Kernel Version: 4.4.0-22-generic
Operating System: Ubuntu 16.04 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 7.795 GiB
Name: hoge
ID: 5HJM:SQPO:3NEM:ZYX7:3LME:WIHB:E3BN:RNMZ:6K4F:JCQ6:UY2Z:CR26
Docker Root Dir: /mnt/disk2/docker3
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Experimental: true
Insecure Registries:
 127.0.0.0/8
```

**Steps to reproduce the issue:**
1. `mkdir /some-new-directory`
2. `ln -s /some-new-directory /var/lib/docker`
3. `systemctl start docker`
4. `docker plugin install tiborvass/no-remove`

**Describe the results you received:**
```
$ docker plugin install tiborvass/no-remove
Plugin "tiborvass/no-remove" is requesting the following privileges:
 - network: [host]
 - mount: [/data]
 - device: [/dev/cpu_dma_latency]
Do you grant the above permissions? [y/N] y
Error response from daemon: rpc error: code = 2 desc = oci runtime error: /var/lib/docker/plugins/0defe3764548da3cb5eea62930717c6cf9f00be9c6e9406f5cf3860bd31bee6d/rootfs is not an absolute path or is a symlink
```

**Describe the results you expected:**
```
$ docker plugin install tiborvass/no-remove
Plugin "tiborvass/no-remove" is requesting the following privileges:
 - network: [host]
 - mount: [/data]
 - device: [/dev/cpu_dma_latency]
Do you grant the above permissions? [y/N] y
$ docker plugin ls
NAME                  TAG                 ACTIVE
tiborvass/no-remove   latest              true
```

**Comment by AkihiroSuda** [comment]
Confirmed that this issue is fixed in 1.12.2-rc1

I think I saw a PR that fixes this issue, but didn't find the actual PR number

**Comment by thaJeztah** [comment]
Thanks @AkihiroSuda, I'll move the milestone. We can link to the correct PR if we find it :-)
